app/models/invite_code_validator.rb

```diff
@@ -0,0 +1,29 @@
+class InviteCodeValidator < ActiveModel::Validator
+  def validate(user)
```

Comment:
Hmmm, good point, but I'm still trying to understand why the current version works but using user_invite_code for both returns a nil:NilClass error when I change the code to 

```javascript
if not_existent?(user_invite_code.invite_code) 
```
rather than

```javascript
if not_existent?(user.invite_code)
```
Shouldn't they be the same? 

(I don't think we can use ```user.invite_code``` for both because ```used_already?``` needs the invite_count, and user doesn't have one.)